src/daft-functions/src/temporal/truncate.rs

```diff
@@ -0,0 +1,78 @@
+use common_error::{DaftError, DaftResult};
+use daft_core::prelude::*;
+use daft_dsl::{
+    functions::{ScalarFunction, ScalarUDF},
+    ExprRef,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct Truncate {
+    interval: String,
+}
+
+#[typetag::serde]
+impl ScalarUDF for Truncate {
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
+
+    fn name(&self) -> &'static str {
+        stringify!($fn_name)
```

Comment:
How is this working outside of a macro with `$fn_name`? 😮

Reply:
`stringify` is literally stringifying it to `"$fn_name"`. `$fn_name` here isn't a variable or macro binding, it's literally a sequence of characters which `stringify` turns into a string.

Reply:
Not sure why we're doing this, however...

Reply:
Maybe `typetag::serde` is doing some internal, black-box magic?

Reply:
good catch, bad copy/paste here